File: lab6/lab6/gen.py
from string import ascii_letters
from random import randint
from tqdm import tqdm

LINES = 10000
LENGTH = 100000

output = ""
letters = len(ascii_letters)

with open("./input.txt", "w") as f:
    for line in tqdm(range(LINES)):
        [f.write(ascii_letters[randint(0, letters-1)]) for letter in range(LENGTH)]
        f.write("\n")

File: lab6/lab6/gen.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>

#define LINES 10000
#define LENGTH 100000
#define PROGRESS_INTERVAL 100

int main() {
    srand(time(NULL));
    const char *ascii_letters = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ";
    FILE *f = fopen("input.txt", "w");

    for (int line = 0; line < LINES; line++) {
        for (int letter = 0; letter < LENGTH; letter++) {
            fputc(ascii_letters[rand() % 52], f);
        }
        fputc('\n', f);

        if (line % PROGRESS_INTERVAL == 0) {
            printf("Progress: %d/%d lines generated\n", line, LINES);
        }
    }

    fclose(f);
    return 0;
}
